### PR TITLE
🔧 Do not rebuild sqlite3 Dependency on Every Dev. Start

### LINKED ISSUE #1307: 🔧 SQLite Module is rebuild on every electron-start-dev execution
#### Describe your issue

When starting the development setup of the BPMN-Studio using `npm run electron-start-dev`, the `sqlite` Module is rebuild everytime. 

This shouldn't really be necessary (at least it wasn't in the past) and is a bit time consuming. 

#### How can others reproduce the issue?

* Run `npm run electron-start-dev`
* See, that the sqlite module is compiled every time

#### Possible solution

Check, if the sqlite module really needs to be compiled on every execution of `npm run electron-start-dev`.

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by robpal94**
---

**Changes:**

1. Currently, the `electron-rebuild` command, which is executed every time someone executes `npm run electron-start-dev`, has the following flags: 
* `--force`
* `--which-module` 

The problem with this is, that the `--force` flag seems to kinda *negate* the `--which-module` Flag. 
This is described in the documentation of the `electron-rebuild` packet: 
> -f, --force                  Force rebuilding modules, **even if we would skip it otherwise**

This Pr removes the `--which-module` flag on the normal `electron-rebuild` shortcut and adds a new shortcut which only sets the `--which module` flag (with the deasync module as a parameter). 

**Issues:**

Closes #1307 

PR: #1316 

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -335,9 +343,28 @@ Die folgenden Skripte, werden in unserem Tooling verwendet:
...
+   Baut alle nativen Abhängigkeiten.
+
+* `test-init`
+
+   Startet die installierte ProcessEngine und den Webdriver, welcher für
+   die End-to-End Tests benötigt wird.
+
+* `test`
+
+   Führt die End-to-End tests für das BPMN - Studio FrontEnd aus.
```
Comment (alexanderkasten) [inline README.md]:
```suggestion
   Führt die End-to-End Tests für das BPMN-Studio Frontend aus.
```

File: README.md [hunk 2]
```diff
@@ -301,6 +301,14 @@ Die folgenden Skripte, werden in unserem Tooling verwendet:
 
    Baut das Aurelia Bundle.
 
+* `reinstall`
+
+  Entfernt die aktuell installierten Abhängigkeiten und installiert diese neu.
+
+* `reset`
+
+  Entfernt alle aktuell installierten Abhängigkeiten
```
Comment (alexanderkasten) [inline README.md]:
```suggestion
  Entfernt alle aktuell installierten Abhängigkeiten.
```

File: README.md [hunk 3]
```diff
@@ -335,9 +343,28 @@ Die folgenden Skripte, werden in unserem Tooling verwendet:
...
+   Startet die installierte ProcessEngine und den Webdriver, welcher für
+   die End-to-End Tests benötigt wird.
+
+* `test`
+
+   Führt die End-to-End tests für das BPMN - Studio FrontEnd aus.
+
+* `test-docker`
+
+   Startet die BPMN - Studio Webanwendung als Docker Container und führt
```
Comment (alexanderkasten) [inline README.md]:
```suggestion
   Startet die BPMN-Studio Webanwendung als Docker Container und führt
```

File: README.md [hunk 4]
```diff
@@ -301,9 +301,17 @@ Die folgenden Skripte, werden in unserem Tooling verwendet:
...
+  Entfernt die aktuell installierten Abhängigkeiten und installiert diese neu.
+
+* `reset`
+
+  Entfernt alle aktuell installierten Abhängigkeiten.
+
 * `start`
 
-   Startet die BPMN-Studio Webanwendung
+   Startet die BPMN-Studio Webanwendung.
```
Comment (alexanderkasten) [inline README.md]:
```suggestion
  Startet die BPMN-Studio Webanwendung.
```

File: README.md [hunk 5]
```diff
@@ -335,9 +343,28 @@ Die folgenden Skripte, werden in unserem Tooling verwendet:
 
    Baut die Electron-Anwendung für Windows.
 
+* `electron-rebuild`
+
+   Baut alle nativen Abhängigkeiten.
```
Comment (alexanderkasten) [inline README.md]:
```suggestion
  Baut alle nativen Abhängigkeiten.
```

File: README.md [hunk 6]
```diff
@@ -335,9 +343,28 @@ Die folgenden Skripte, werden in unserem Tooling verwendet:
...
+
+* `test-init`
+
+   Startet die installierte ProcessEngine und den Webdriver, welcher für
```
Comment (alexanderkasten) [inline README.md]:
```suggestion
  Startet die installierte ProcessEngine und den Webdriver, welcher für
```

File: README.md [hunk 7]
```diff
@@ -335,9 +343,28 @@ Die folgenden Skripte, werden in unserem Tooling verwendet:
...
+   die End-to-End Tests benötigt wird.
```
Comment (alexanderkasten) [inline README.md]:
```suggestion
  die End-to-End Tests benötigt wird.
```

File: README.md [hunk 8]
```diff
@@ -335,9 +343,28 @@ Die folgenden Skripte, werden in unserem Tooling verwendet:
...
+
+* `test`
+
+   Führt die End-to-End Tests für das BPMN-Studio Frontend aus.
```
Comment (alexanderkasten) [inline README.md]:
```suggestion
  Führt die End-to-End Tests für das BPMN-Studio Frontend aus.
```

File: README.md [hunk 9]
```diff
@@ -335,9 +343,28 @@ Die folgenden Skripte, werden in unserem Tooling verwendet:
...
+
+* `test-docker`
+
+   Startet die BPMN-Studio Webanwendung als Docker Container und führt
```
Comment (alexanderkasten) [inline README.md]:
```suggestion
  Startet die BPMN-Studio Webanwendung als Docker Container und führt
```

File: README.md [hunk 10]
```diff
@@ -335,9 +343,28 @@ Die folgenden Skripte, werden in unserem Tooling verwendet:
...
+   die End-to-End Tests gegen der im Container laufenden Instanz der
```
Comment (alexanderkasten) [inline README.md]:
```suggestion
  die End-to-End Tests gegen der im Container laufenden Instanz der
```

File: README.md [hunk 11]
```diff
@@ -335,9 +343,28 @@ Die folgenden Skripte, werden in unserem Tooling verwendet:
...
+   BPMN-Studio Webanwendung aus.
```
Comment (alexanderkasten) [inline README.md]:
```suggestion
  BPMN-Studio Webanwendung aus.
```

File: package.json [hunk 12]
```diff
@@ -22,13 +22,13 @@
     "test-crossbrowser": "au e2eCrossbrowser",
     "lint": "tslint --project .",
     "electron-start-dev": "npm run electron-rebuild && npm run build && electron electron_app/electron.js",
-    "electron-rebuild": "electron-rebuild --force --which-module deasync",
+    "electron-rebuild": "electron-rebuild",
```
Comment (alexanderkasten) [inline package.json]:
```suggestion
```

File: package.json [hunk 13]
```diff
@@ -22,13 +22,13 @@
...
     "electron-build-windows": "npm run electron-rebuild && npm run build && build --publish never --windows",
-    "electron-build-macos": "npm run build && build --publish never --macos",
-    "electron-build-linux": "npm run build && build --publish never --linux",
+    "electron-build-macos": "npm run electron-rebuild && npm run build && build --publish never --macos",
```
Comment (alexanderkasten) [inline package.json]:
```suggestion
    "electron-build-macos": "electron-rebuild && npm run build && build --publish never --macos",
```

File: package.json [hunk 14]
```diff
@@ -22,13 +22,13 @@
...
+    "electron-build-linux": "npm run electron-rebuild && npm run build && build --publish never --linux",
```
Comment (alexanderkasten) [inline package.json]:
```suggestion
    "electron-build-linux": "electron-rebuild && npm run build && build --publish never --linux",
```

File: package.json (hunk 12 above)
Comment (robpal94) [reply package.json]: @Paulomart and I discussed, that we want to define the call to `electron-rebuild`  as a dedicated shortcut script in the `package.json` file.

File: package.json (hunk 13 above)
Comment (robpal94) [reply package.json]: See above

File: package.json (hunk 14 above)
Comment (robpal94) [reply package.json]: See above

File: package.json [hunk 15]
```diff
@@ -21,14 +21,13 @@
     "test-docker": "bash test/start_docker_bundle.sh && au e2eDocker",
     "test-crossbrowser": "au e2eCrossbrowser",
     "lint": "tslint --project .",
-    "electron-start-dev": "npm run electron-rebuild && npm run build && electron electron_app/electron.js",
-    "electron-rebuild": "electron-rebuild --force --which-module deasync",
-    "electron-build-windows": "npm run electron-rebuild && npm run build && build --publish never --windows",
-    "electron-build-macos": "npm run build && build --publish never --macos",
-    "electron-build-linux": "npm run build && build --publish never --linux",
+    "electron-start-dev": "electron-rebuild && npm run build && electron electron_app/electron.js",
```
Comment (alexanderkasten) [inline package.json]: Ist das jetzt richtig so?
Im post install werden schon alle Pakete rebuilded. Und electron-rebuild ist nur für den Jenkins oder die Native App wichtig, also wenn man das Studio gezielt für eine Platform baut.

Comment (robpal94) [reply package.json]: `electron-rebuild` sollte selber entscheiden, welche Pakete evtl. neu gebaut werden müssen. In der Regel stimmt es schon, dass nach dem `install` keine native Abhängigkeit mehr gebaut werden muss. Allerdings würde das in der Regel auch `electron-rebuild` merken und die build steps dann dementsprechend überspringen. 
Sollte jedoch aus welchen Gründen immer eine Dependency neu gebaut werden müssen, würde `electron-rebuild` das von sich aus tun.

**Pull request merged**